transport-protocol/src/custom.rs

```diff
@@ -23,35 +35,157 @@ pub enum TransportProtocolError {
     ResponseStatusError(i32),
     #[error(display = "TransportProtocol: TryIntoError: {}.", _0)]
     TryIntoError(#[error(source)] std::num::TryFromIntError),
+    #[error(
+        display = "TransportProtocol: Session {:?}: Buffer is partial, still waiting for chunks",
+        _0
+    )]
+    PartialBuffer(u32),
+    #[error(display = "TransportProtocol: Session {:?}: Length prefix missing", _0)]
+    MissingLengthPrefix(u32),
+    #[error(
+        display = "TransportProtocol: Session {:?}: Incoming buffer not found",
+        _0
+    )]
+    BufferNotFound(u32),
+    #[error(
+        display = "TransportProtocol: Session {:?}: Can't get mutex for incoming buffer",
+        _0
+    )]
+    MutexError(u32),
 }
 type TransportProtocolResult = Result<std::vec::Vec<u8>, TransportProtocolError>;
 
+/// General documentation on the network stack (TODO: move it somewhere else)
```

Comment:
Is this TODO in the issue tracker so it is not forgotten?

Reply:
Done